app/src/main/java/fr/free/nrw/commons/category/CategoriesModel.kt

```diff
@@ -27,30 +27,41 @@ class CategoriesModel @Inject constructor(
     private var selectedExistingCategories: MutableList<String> = mutableListOf()
 
     /**
-     * Returns if the item contains an year
+     * Returns true if the item contains an year which should be ignored
      * @param item
      * @return
      */
     fun containsYear(item: String): Boolean {
```

Comment:
At this point, I think it would be better to rename this function to something better like `isFilterableCategory` / `isSpammyCategory` (we could find a better name than that).